internal/agent/logic.go

```diff
@@ -415,78 +414,99 @@ func (a *defaultAgent) ecosystemExpertise() []string {
 	}
 }
 
-func (a *defaultAgent) historyContext(ctx context.Context) []string {
+// executionDetails about the previous execution
+type executionDetails struct {
+	Iteration    schema.AgentIteration
+	Instructions rebuild.Instructions
+	Dockerfile   string
+}
+
+func (a *defaultAgent) dockerfile(ctx context.Context, obliviousID string) (string, error) {
+	if obliviousID == "" {
+		return "", errors.New("no oblivious ID provided")
+	}
+	meta, err := rebuild.NewGCSStoreFromClient(context.WithValue(ctx, rebuild.RunID, obliviousID), a.deps.GCSClient, fmt.Sprintf("gs://%s", a.deps.MetadataBucket))
+	if err != nil {
+		return "", err
+	}
+	r, err := meta.Reader(ctx, rebuild.DockerfileAsset.For(a.t))
+	if err != nil {
+		return "", errors.Wrap(err, "opening dockerfile")
+	}
+	defer r.Close()
+	data, err := io.ReadAll(r)
+	if err != nil {
+		return "", errors.Wrap(err, "reading dockerfile")
+	}
+	return string(data), nil
+}
+
+func (a *defaultAgent) execDetails(ctx context.Context, iteration *schema.AgentIteration) *executionDetails {
+	d := &executionDetails{Iteration: *iteration}
+	if iteration.Strategy == nil {
+		return d
+	}
+	s, err := iteration.Strategy.Strategy()
+	if err != nil {
+		return d
+	}
+	// Instructions
```

Comment:
want to fold these onto trailing comments on the line below? you also might add a bit more here to make this more than a word